flink-connector-bigquery/src/test/java/com/google/cloud/flink/bigquery/common/utils/BigQueryPartitionTest.java

```diff
@@ -0,0 +1,491 @@
+/*
+ * Copyright (C) 2023 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.flink.bigquery.common.utils;
+
+import org.apache.flink.shaded.guava30.com.google.common.collect.Lists;
+
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import com.google.cloud.flink.bigquery.fakes.StorageClientFaker;
+import com.google.cloud.flink.bigquery.services.PartitionIdWithInfo;
+import com.google.cloud.flink.bigquery.services.PartitionIdWithInfoAndStatus;
+import com.google.cloud.flink.bigquery.services.TablePartitionInfo;
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.Optional;
+
+/** */
+public class BigQueryPartitionTest {
```

Comment:
This file is also hard to review due to the move. can we move it at the end after reviewing is complete?